skopt/optimizer/optimizer.py

```diff
@@ -0,0 +1,248 @@
+        which to evluate the objective. This point can be retrieved by calling
+        `ask()`.
+
+        To add several observations as a batch without fitting a new model
+        set `fit_model` to False.
+
+        * `x` [list]:
+            Point at which objective was evaluated.
+        * `y` [scalar]:
+            Value of objective at `x`.
+        * `fit_model` [bool, default=True]
+            Fit a model to observed evaluations of the objective.
+        """
+        self.Xi.append(x)
+        self.yi.append(y)
+
+        if fit_model:
+            transformed_bounds = np.array(self.space.transformed_bounds)
+            est = clone(self.base_estimator)
+
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                est.fit(self.space.transform(self.Xi), self.yi)
+
+            self.models.append(est)
+
+            # Find the minimum of the acquisition function by randomly sampling
+            # points from the space
+            if self.acq_optimizer == "sampling":
+                X = self.space.transform(self.space.rvs(
+                    n_samples=self.n_points, random_state=self.rng))
+                values = _gaussian_acquisition(
+                    X=X, model=est,  y_opt=np.min(self.yi),
+                    acq_func=self.acq_func, xi=self.xi, kappa=self.kappa)
+                next_x = X[np.argmin(values)]
+
+            # Use BFGS to find the mimimum of the acquisition function, the
+            # minimization starts from `n_restarts_optimizer` different points
+            # and the best minimum is used
+            elif self.acq_optimizer == "lbfgs":
+                best = np.inf
+                x0 = self.space.transform(
+                    self.space.rvs(n_samples=self.n_restarts_optimizer,
+                                   random_state=self.rng))
+
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    jobs = (delayed(fmin_l_bfgs_b)(
+                        gaussian_acquisition_1D, x,
+                        args=(est, np.min(self.yi), self.acq_func,
+                              self.xi, self.kappa),
+                        bounds=self.space.transformed_bounds,
+                        approx_grad=False,
+                        maxiter=20) for x in x0)
+                    results = self.parallel(jobs)
+
+                cand_xs = np.array([r[0] for r in results])
+                cand_acqs = np.array([r[1] for r in results])
+                best_ind = np.argmin(cand_acqs)
+                a = cand_acqs[best_ind]
+
+                if a < best:
+                    next_x, best = cand_xs[best_ind], a
+
+            # lbfgs should handle this but just in case there are
+            # precision errors.
+            next_x = np.clip(
+                next_x, transformed_bounds[:, 0], transformed_bounds[:, 1])
+            # XXX why the need for [0] at the end?
```

Comment:
`inverse_transform` operates at the level of an array of points, while here we only want to transform one point. That is why we reshape and then select element 0.